sources/web/datalab/polymer/components/datalab-app/datalab-app.ts

```diff
@@ -115,6 +134,19 @@ class DatalabAppElement extends Polymer.Element {
     }
   }
 
+  _queryParamsChanged() {
+    const queryParams = (this.queryParams || {}) as {[key: string]: string};
+    const fileParamName = 'file';
+    const fileParam = queryParams[fileParamName] || '';
+    if (fileParam !== this.fileId) {
+      this.fileId = fileParam;
```

Comment:
Instead of this, why don't we set the active page element's `fileId`, and call this method on page change event? This saves the `file-browser` element from having to figure out if it's active/visible or now.

Reply:
I think the active vs visible are two separate questions, as it seems to get called when the page is active not not yet visible. We would want to call it from more than just the page-change event, since the fileId can change without the page changing. I think it may be possible to do it the way you are suggesting, but yes, I'd rather commit this now and maybe make that change later.